aten/src/ATen/native/nested/README.md

```diff
@@ -0,0 +1,63 @@
+# NestedTensors
+So you decided to look at the source code.. let's give you a quick overview of the codebase.
+
+## NestedTensor Data Structure
+
+NestedTensors are a generalization of torch Tensors which eases working with data of different shapes and lengths. They are primarily used to represent a list of N tensors, where each tensor in the list (referred to as a tensor_component) has the same number of dimensions. The tensor_components are flattened and combined into a single NestedTensor, which includes the information required to reconstruct the original tensor_components:
+
+- nested_size_tensor_: 2d tensor of n_tensor_components x n_dims
+- nested_stride_tensor_: 2d tensor of n_tensor_components x n_dims
+- storage_offsets_: 1d tensor of offsets corresponding to the start position of each tensor component
+- storage_: The storage object that contains the flattened tensor_components (defined on c10::TensorImp)
+
+NestedTensors inherit from c10::TensorImpl whose definition can be found here: [NestedTensorImpl.h](../../NestedTensorImpl.h).
+
+When constructing a NestedTensor in C++ you will likely not be using the NestedTensorImpl constructor directly but using the `wrap_buffer` function defined in [NestedTensorUtils.h](NestedTensorUtils.h). This is a thin wrapper around the NestedTensorImpl constructor that ensures that the input tensor is contiguous. This is because when constructing a NestedTensor from a dense tensor we create a shallow copy of the input's Storage object and the if the input tensor did not satisfy `tensor.numel() == tensor.storage.numel()` this could lead to undefined behavior.
+
+##  Code Structure
+
+The NestedTensor code is split into two parts: the C++ code and the Python code. The C++ code is located in [aten/src/ATen/native/nested](.) and the Python code is located in [torch/nested/__init__.py](/torch/nested/__init__.py). The C++ code is split into the following files:
```

Comment:
nit: (.)

Reply:
that just redirects you to the the current directory which is where the readme is defined